### PR TITLE
Task/w 001828 write to integration object

### PR DESCRIPTION
**Pull request by jhulbert-va**
[W-001828]()

* Create a New Integration Object Per Day Ran (Success and Fail)
* Updated Deletes to make sure all records are checked (we might want to add the is deleted flag back in my opinion)

### CODE REVIEW

File: src/main/app/imdb-salesforce.xml [hunk 1]
```diff
@@ -282,54 +288,87 @@ mapper.sfResponseMapper(
...
-		updateFailures = updateFailures + 1;
-	}
-}
+        <dw:transform-message doc:name="Determine Record Metrics &amp; Combine Change Types">
+            <dw:set-variable variableName="recordMetrics"><![CDATA[%dw 1.0
+%output application/java
 
-deleteSuccess = 0;
-deleteFailures = 0;
+%function getRecordMetrics(records)
```
Comment (jgarnhart-va) [inline src/main/app/imdb-salesforce.xml]: Nice solution here. :)

File: src/main/app/imdb-salesforce.xml [hunk 2]
```diff
@@ -282,54 +288,87 @@ mapper.sfResponseMapper(
...
+		when flowVars.recordMetrics != null otherwise 0)
+%var failureTotal = (addMetrics(flowVars.recordMetrics[0].failures,
+		flowVars.recordMetrics[1].failures,
+		flowVars.recordMetrics[2].failures)
+		when flowVars.recordMetrics != null otherwise 0)
+---
+[{
+	Application_Name__c: "Mule_DMC_IMDB_Integration",
+	Direction__c: "From IMDB",
+	//Josephs File Name
```
Comment (jgarnhart-va) [inline src/main/app/imdb-salesforce.xml]: Could we change this comment to something like "Data errors file name"?

File: src/main/resources/config-tst.properties [hunk 3]
```diff
@@ -1,20 +1,21 @@
...
+https.private.port=![IyUuqyCCLac6mGO6bz3/xA==]
+
+oauth.https.host=![LXEVnMtCETsc2Pwgc1vzRjgvWeaf7vXxajZ7Q/PPvm8=]
+imdb.https.host=![f1xOKIDzYP1kd9BqV6SliPAa+StGJKqqlgxg2/6PXPU=]
+imdb.https.port=![Xb7+CtZLNHA58dfz9oQ7jw==]
+imdb.https.basepath=![jqiuslN+FXfcO3UGjhYw2w==]
+
+cron.scheduler.expression=![vhqx9nAMRRfMcMeTm+pDNQ==]
+maximum.records.size=![4UkfYvQtoHYDw08Et+FEFA==]
+integration.poc=![3/01e8i1WTdRNB4jN8euQQ==]
```
Comment (jgarnhart-va) [inline src/main/resources/config-tst.properties]: Lets add `integration.poc` to config-dev.properties as well.

File: src/main/app/imdb-salesforce.xml [hunk 4]
```diff
@@ -282,54 +288,87 @@ mapper.sfResponseMapper(
...
+			otherwise "NoRecords" when totalRecords == 0
+			otherwise "Success" when failureTotal == 0
+			otherwise "SuccessWithFailures" when failureTotal > 0
+			otherwise "Undetermined"),
+	Interface_Name__c: "IMDB",
+	Job_Run_Time__c: now as :datetime,
+	Point_Of_Contact__c: p('integration.poc'),
+	Source__c: "IMDB",
+	Total_Records_Processed__c: totalRecords,
+	Total_Records_Successful__c: (addMetrics(flowVars.recordMetrics[0].successes,
```
Comment (jgarnhart-va) [inline src/main/app/imdb-salesforce.xml]: Curious why success count is determined above the way total and failures are?

**Comment by jgarnhart-va** [comment]
I ran a manual job when _not_ connected to VA VPN to force a failure trying to get imdb access token. Failed as expected and a nice integration job was created:

    {
        "Total_Records_Errors__c": "0.0",
        "Application_Name__c": "Mule_DMC_IMDB_Integration",
        "Id": null,
        "type": "VBA_ERP_Integration_Job_Status__c",
        "Job_Run_Time__c": "2019-03-05T02:17:36.000Z",
        "Total_Records_Processed__c": "0.0",
        "Total_Records_Successful__c": "0.0",
        "Name": "dmcImdbIntegration_NoDateFound_Fail"
    }

Nice work here :)

File: src/main/app/imdb-salesforce.xml (hunk 4 above)
Comment (jhulbert-va) [reply src/main/app/imdb-salesforce.xml]: total and failures are being used to detmine "Success" or "Success w/ Failures" I can add this to a var to keep it uniform.

**Pull request merged**